lib/authentication/secure_storage/json_credential_manager.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const path = require('path');
+const Logger = require('../../logger');
+const fs = require('fs');
+const os = require('os');
+const Util = require('../../util');
+
+function JsonCredentialManager() {
+
+  this.userinfo = os.userInfo();
+
+  this.getTokenDir = function () {
+    let tokenDir = process.env.SF_TEMPORARY_CREDENTIAL_CACHE_DIR;
+
+    if (!Util.exists(tokenDir)) {
+      tokenDir = os.homedir();
+    } else {
+      Logger.getInstance().info(`The credential cache directory is configured by the user. The token will be saved at ${tokenDir}`);
+    }
+
+    if (!Util.exists(tokenDir)) {
+      Logger.getInstance().error(`Temporary credential cache directory is invalid, and the driver is unable to use the default location(home). 
+      Please assign the enviroment variable value SF_TEMPORARY_CREDENTIAL_CACHE_DIR to enable the default credential manager.`);
+      return null; 
+    }
+    const tokenCacheFile = path.join(tokenDir, 'temporary_credential.json');
+
+    if (fs.existsSync(tokenCacheFile)) {
+      try {
+        fs.accessSync(tokenCacheFile,  fs.constants.R_OK | fs.constants.W_OK);
+      } catch (err) {
+        Logger.getInstance().info(`You do not have permission to the credential file: ${tokenCacheFile}.`);
+        return null;
+      }
+    }
+    Logger.getInstance().info('Cache directory: ', tokenCacheFile);
+    return tokenCacheFile;
+  };
+
+  this.tokenDir = this.getTokenDir();
+
+  this.readJsonCredentialFile = function () {
+    if (this.tokenDir === null) {
+      return null;
```

Comment:
should we log before returning null? should we throw the exception? my question about null vs exception is also for following code

Reply:
We do not need to log some messages at this point because other error messages are logged already when the driver validates the path on this.getTokenDir. if the tokenDir is null  Also, I am confused about whether we need to throw an exception or disable the credential cache and keep running the driver. Please let me know :)

Reply:
We are using this function e.g. 
```
const jsonCredential = this.readJsonCredentialFile();
return jsonCredential[key] || null;
```
so here there will be the error when json credentials cannot be read - it's better to throw the exception and handle it in upper layers (or we should return an empty object)

Reply:
As I replaced the sync functions to async functions, the error will be handled on the upper layer. Please have a look. Thank you.